File: solutions/c/pangram/1/pangram.c
#include "pangram.h"
#include <ctype.h>
#include <string.h>
#include <stdbool.h>

bool incidence(const char* sentence, char character) {
    if (sentence == NULL) {
        return false;
    }
    for(size_t i = 0; sentence[i] != '\0'; i++) {
        if(tolower(character) == tolower(sentence[i])) {
            return true;
            break;
        }
    }
    return false;
}

bool is_pangram(const char *sentence) {
    if(sentence == NULL || strlen(sentence) < 26) {
        return false;
    }
    for(char c = 'a'; c <= 'z'; c++) {
        if(incidence(sentence, c) == false) {
            return false;
        }
    }
    return true;
}

File: solutions/haskell/pangram/1/src/Pangram.hs
module Pangram (isPangram) where

import Data.Char(toLower,isLetter)

isPangram :: String -> Bool
isPangram text = all (`elem` loweredLetters) ['a'..'z']
  where
    loweredText = map toLower text
    loweredLetters = filter isLetter loweredText
